publish.js

```diff
@@ -0,0 +1,270 @@
+    }
+};
+
+/**
+ * Updates the package.json version to the specified version
+ * This does not commit the change, it will only modify the file.
+ *
+ * @param version {String} - the version to set in the package.json file
+ * @param options {Object} - e.g. {"versionCmd": "npm version --no-git-tag-version ${version}"}
+ */
+publish.setVersion = function (version, options) {
+    var cmdTemplate = options.versionCmd || defaults.versionCmd;
+    var cmdStr = es6Template(cmdTemplate, {
+        version: version
+    });
+    publish.execSync(cmdStr);
+};
+
+/**
+ * Calculates the current dev version of the package
+ *
+ * @param options {Object} - e.g. {"rawTimestampCmd": "git show -s --format=%ct HEAD", "revisionCmd": "git rev-parse --verify --short HEAD", "devVersion": "${version}.${timestamp}.${revision}"}
+ * @returns {String} - the current dev version number
+ */
+publish.getDevVersion = function (options) {
+    var rawTimestamp = publish.execSync(options.rawTimestampCmd || defaults.rawTimestampCmd);
+    var timestamp = publish.convertToISO8601(rawTimestamp);
+    var revision = publish.execSync(options.revisionCmd || defaults.revisionCmd);
+    var devVersionTemplate = options.devVersion || defaults.devVersion;
+    var newStr = es6Template(devVersionTemplate, {
+        version: pkg.version,
+        timestamp: timestamp,
+        revision: revision
+    });
+    return newStr;
+};
+
+/**
+ * Publishes the module to npm using the current version number in pacakge.json.
+ * If isTest is specified, it will instead create a tarball in the local directory.
+ *
+ * @param isTest {Boolean} - indicates if this is a test run or not
+ * @param options {Object} - e.g. {"packCmd": "npm pack", "publishCmd": "npm publish"}
+ */
+publish.pubImpl = function (isTest, options) {
+    if (isTest) {
+        // create a local tarball
+        var packCmd = options.packCmd || defaults.packCmd;
+        publish.execSync(packCmd);
+    } else {
+        // publish to npm
+        var publishCmd = options.publishCmd || defaults.publishCmd;
+        publish.execSync(publishCmd);
+    }
+};
+
+/**
+ * Tags the specified version with the specified dist-tag
+ * If it is a test run, the tag command will be output to the console.
+ *
+ * @param isTest {Boolean} - indicates if this is a test run or not
+ * @param version {String} - a string idicating which version to tag
+ * @param tag {String} - the dist-tag to apply
+ * @param options {Object} - e.g. {"distTagCmd": "npm dist-tag add infusion@${version} ${tag}"}
+ */
+publish.tag = function (isTest, version, tag, options) {
+    var cmdTemplate = options.distTagCmd || defaults.distTagCmd;
+    var cmdStr = es6Template(cmdTemplate, {
+        version: version,
+        tag: tag
+    });
+    if (isTest) {
+        publish.log("tag command: " + cmdStr);
+    } else {
+        publish.execSync(cmdStr);
+    }
+};
+
+/**
+ * Resets the current workspace.
+ * This will clear out any git tracked changes.
```

Comment:
Update this comment and example based on new default, e.g. "This will undo any changes made by this tool to package,.json"